be/src/olap/rowset/segment_v2/column_reader.cpp

```diff
@@ -210,6 +210,10 @@ Status ColumnReader::read_page(const ColumnIteratorOptions& iter_opts, const Pag
     opts.type = iter_opts.type;
     opts.encoding_info = _encoding_info;
     opts.io_ctx = iter_opts.io_ctx;
+    // index page should not pre decode
+    if (iter_opts.type == INDEX_PAGE) {
```

Comment:
maybe add some regression test will more better ?

Reply:
it's a inner bug and logic here is two mistakes leads to running correnct, we already have case.